[docs] Update enabling developer mode instructions on iOS devices (#22648)
expo/expo: docs/pages/guides/ios-developer-mode.mdx

import ImageSpotlight from '~/components/plugins/ImageSpotlight';
import { Step } from '~/ui/components/Step';

> This does not apply to builds signed using enterprise provisioning, or to any builds installed on an iOS Simulator.

Devices running iOS 16 and above need to enable OS-level **Developer Mode** setting before they can run [internal distribution](/build/internal-distribution) builds (including those built with EAS) or local development builds after installing them on the device.

## Prerequisites

> The instructions specified below need to be followed once per device. **You'll also need to install your development build on your device before you can enable the Developer Mode.** After creating your first build, follow the instructions from the Expo dashboard to install it on your iOS device.

Author: ```suggestion
The instructions specified below need to be followed once per device. **You'll also need to install your development build on your device before you can enable the Developer Mode.** When the build is created, follow the instructions from the Expo dashboard to install it on your iOS device.
```

import ImageSpotlight from '~/components/plugins/ImageSpotlight';
import { Step } from '~/ui/components/Step';

> This does not apply to builds signed using enterprise provisioning, or to any builds installed on an iOS Simulator.

Devices running iOS 16 and above need to enable OS-level **Developer Mode** setting before they can run [internal distribution](/build/internal-distribution) builds (including those built with EAS) or local development builds after installing them on the device.

## Prerequisites

The instructions specified below need to be followed once per device. **You'll also need to install your development build on your device before you can enable the Developer Mode.**

## Enable Developer Mode

<Step label="1">

> After creating your first development build, follow the instructions from the Expo dashboard to install it on your iOS device.

Reviewer: I often share a link to this section with individuals interested in testing through internal distribution, many of whom are not developers. I believe the phrase "After creating your first development build" might be confusing to them. Since this section primarily focuses on enabling Developer Mode on iOS, I suggest either removing or modifying that particular phrase.

Author: If they are testing internal distribution,  they'll have to install it on their device, and for that reason this section talks about what to do when they're installing the development build.

Maybe we can improve the phrasing here.

Author: Removed these instructions from step 1 and added them to prerequisites.
